# SALES DATA ANALYSIS

**AI Academia — Python with Data Science Internship**

**Name:** Rohit Panda  
**Date:** 15 October 2026

This project analyses retail sales data to identify seasonal trends, top products, and predict sales using machine learning models.

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('sales_data_final.csv')

print(df.shape)
print(df.columns.tolist())
print(df.head())

(1208, 7)
['Date', 'Product', 'Category', 'Region', 'Quantity', 'Unit_Price', 'Sales']
         Date  Product     Category Region  Quantity  Unit_Price      Sales
0  2025-01-01  Printer       Office   East         7       15000  103461.87
1  2025-01-01    Mouse  Accessories  North        10        1500   14635.45
2  2025-01-01   Tablet  Electronics  North         3       25000   73203.26
3  2025-01-01  Printer       Office  North        10       15000  154235.97
4  2025-01-02   Laptop  Electronics   West        14       60000  787737.60


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1208 entries, 0 to 1207
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Date        1208 non-null   str    
 1   Product     1208 non-null   str    
 2   Category    1208 non-null   str    
 3   Region      1202 non-null   str    
 4   Quantity    1208 non-null   int64  
 5   Unit_Price  1208 non-null   int64  
 6   Sales       1202 non-null   float64
dtypes: float64(1), int64(2), str(4)
memory usage: 66.2 KB


In [3]:
df.isnull().sum()

Date          0
Product       0
Category      0
Region        6
Quantity      0
Unit_Price    0
Sales         6
dtype: int64

In [4]:
df['Region'] = df['Region'].fillna(df['Region'].mode()[0])
df['Sales'] = df['Sales'].fillna(df['Sales'].median())

print(df.isnull().sum())

Date          0
Product       0
Category      0
Region        0
Quantity      0
Unit_Price    0
Sales         0
dtype: int64


In [5]:
print("Duplicate rows:", df.duplicated().sum())

Duplicate rows: 8


In [6]:
df = df.drop_duplicates()

print("Duplicate rows after cleaning:", df.duplicated().sum())
print("New shape:", df.shape)

Duplicate rows after cleaning: 0
New shape: (1200, 7)


In [7]:
df['Date'] = pd.to_datetime(df['Date'])

print(df.dtypes)

Date          datetime64[us]
Product                  str
Category                 str
Region                   str
Quantity               int64
Unit_Price             int64
Sales                float64
dtype: object


In [8]:
df.describe()

,Date,Quantity,Unit_Price,Sales
count,1200,1200.000000,1200.000000,1.200000e+03
mean,2025-07-01 12:01:12,7.790833,18617.500000,1.478356e+05
min,2025-01-01 00:00:00,1.000000,1500.000000,1.210050e+03
25%,2025-04-01 18:00:00,4.000000,3000.000000,1.970319e+04
50%,2025-07-01 12:00:00,8.000000,15000.000000,5.378546e+04
75%,2025-09-30 06:00:00,12.000000,25000.000000,1.914488e+05
max,2025-12-31 00:00:00,15.000000,60000.000000,1.120239e+06
std,NaN,4.326008,20541.429332,2.093755e+05


In [9]:
df.nunique()

Date           365
Product          8
Category         3
Region           4
Quantity        15
Unit_Price       8
Sales         1195
dtype: int64

In [10]:
# Day 19 - Business Insights

# 1. Category revenue percentage
category_sales = df.groupby('Category')['Sales'].sum()
category_pct = (category_sales / df['Sales'].sum() * 100).sort_values(ascending=False)

# 2. Monthly average sales
monthly_avg = df.groupby(df['Date'].dt.to_period('M'))['Sales'].mean()
overall_avg = df['Sales'].mean()
peak_month = monthly_avg.idxmax()
peak_month_value = monthly_avg.max()
above_avg_pct = (peak_month_value / overall_avg - 1) * 100

# 3. Top and bottom 3 products
product_sales = df.groupby('Product')['Sales'].sum().sort_values(ascending=False)

# 4. Monthly sales and R-squared
monthly_sales = df.groupby(df['Date'].dt.month)['Sales'].sum()

from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

X = monthly_sales.index.values.reshape(-1, 1)
y = monthly_sales.values

model = LinearRegression()
model.fit(X, y)

r2 = r2_score(y, model.predict(X))

print("Category Revenue %:")
print(category_pct)

print("\nHighest Average Sales Month:")
print(peak_month, peak_month_value)
print("Above overall average:", round(above_avg_pct, 2), "%")

print("\nTop 3 Products:")
print(product_sales.head(3))

print("\nBottom 3 Products:")
print(product_sales.tail(3))

print("\nR-squared:", round(r2, 3))

Category Revenue %:
Category
Electronics    83.411060
Office          8.548361
Accessories     8.040579
Name: Sales, dtype: float64

Highest Average Sales Month:
2025-12 199761.78575757574
Above overall average: 35.12 %

Top 3 Products:
Product
Laptop     1.003479e+08
Tablet     2.612519e+07
Monitor    2.150047e+07
Name: Sales, dtype: float64

Bottom 3 Products:
Product
Headphones    4125347.81
Keyboard      3611193.73
Mouse         2062100.61
Name: Sales, dtype: float64

R-squared: 0.122


## KEY FINDINGS

### Finding 1
**Finding:** Electronics generates 83.41% of total revenue, making it the dominant category.  
**Meaning:** The business depends heavily on Electronics for revenue generation.  
**Action:** Maintain strong inventory and marketing support for Electronics products.

### Finding 2
**Finding:** December has the highest average monthly sales at ₹199,761.79, which is 35.12% above the overall average.  
**Meaning:** December shows the strongest sales performance and may indicate increased customer demand.  
**Action:** Increase inventory and promotional activities before December.

### Finding 3
**Finding:** The top 3 products by sales are Laptop (₹100.35M), Tablet (₹26.13M), and Monitor (₹21.50M).  
**Meaning:** These products contribute the most to overall sales revenue.  
**Action:** Prioritize stock availability and marketing for these high-performing products.

### Finding 4
**Finding:** The bottom 3 products are Headphones (₹4.13M), Keyboard (₹3.61M), and Mouse (₹2.06M).  
**Meaning:** These products generate considerably less revenue than the top-performing products.  
**Action:** Review their pricing, promotions, and customer demand before increasing inventory.

### Finding 5
**Finding:** The R-squared score for the seasonal sales pattern is 0.122.  
**Meaning:** The seasonal model explains only about 12.2% of the variation in sales, indicating a weak seasonal relationship.  
**Action:** The business should consider other factors such as product, quantity, price, and region when planning sales.

## RECOMMENDATIONS

1. Increase inventory and promotional campaigns for Electronics products, especially Laptop, Tablet, and Monitor, because they are the strongest revenue generators.

2. Prepare additional stock and marketing campaigns before December while reviewing the pricing and demand of low-performing products such as Headphones, Keyboard, and Mouse.